### DISCUSSION TITLE
Unable To Subscribe To $SYS Topics

### DISCUSSION THREAD

**Discussion by venkatac-ayla** [emqx/emqx#13491] [Q&A] [answered]
While trying to subscribe to $SYS topics we receive the following error. Any idea what is happening

2024-07-10T12:59:36.662643+00:00 [debug] tag: AUTHZ, clientid: dnc-7408a746-f757-4017-8bb2-d351c3e554c4, msg: authorization_module_nomatch, peername: 10.160.40.243:56048, username: testUser, topic: $SYS/brokers/+/clients/+/connected, action: SUBSCRIBE(Q2), module: emqx_authz_client_info

**Comment by ieQu1** [comment]
Default authorization settings prevent clients with IP not equal to the localhost from subscribing to SYS topics. You can adjust authorization settings to enable it.

**Reply by venkatac-ayla** [reply]
But the ACL that is configured is as follow

{allow, {username, {re, "^dashboard$"}}, subscribe, ["$SYS/#"]}.

{allow, {ipaddr, "127.0.0.1"}, all, ["$SYS/#", "#"]}.

{deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.

%% other rule
{allow, {user, "testUser"}, all, ["$SYS/#",  "#"]}.

{allow, {user, "testUser"}, subscribe, ["EVENTS/${clientid}"]}.

**Reply by ieQu1** [reply, marked as answer]
Well, and it works exactly as configured:

```
{allow, {ipaddr, "127.0.0.1"}, all, ["$SYS/#", "#"]}.
{deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.
```

clientid: dnc-7408a746-f757-4017-8bb2-d351c3e554c4, msg: authorization_module_nomatch, peername: **10.160.40.243**:56048,

ACl rules are processed in from top to bottom, once EMQX encounters the second line (`{deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.`), which matches the `testUser`, it stops processing and denies subscription. 

You need to move other ACLs up the list.
